### ISSUE TITLE
grammar start rule

### ISSUE THREAD

**Issue by rns** [rns/kollos-luif-doc#10]
Currently, the produce operation for lexical grammar `~` doesn't allow specifying the grammar name. 

It is ok now, as initally only g1 and l0 will be supported, but needs to be addressed in future.

**Comment by jeffreykegler** [comment]
We should have a way of specifying a "default grammar" for the produce operator (`::=`), which will not affect the tilde operator ("~").  For that there should be a  'tilde default grammar".

I'm not sure if I said or implied that anywhere.  If not, we should add it.

**Comment by rns** [comment]
Perhaps we can make `:grammar_name:=` produce-op start a grammar (and declare a start symbol, explicitely or via adverbs), e.g. 

`start_symbol :grammar_name:= symbol_name` or 

`lhs :grammar_name:= rhs, start = start_symbol, lexical = lexical_grammar_name`

Then, after such statement, 
- all bare structural (`::=`) rules will belong in the structural grammar named `grammar_name` and keyed under it in KIR table.
- all bare lexical (`~`) rules will be linked to `grammar_name` and keyed under, e.g. `grammar_name_lex` key in KIR table.

**Comment by rns** [comment]
Some details on grammar start rule are in #28.

**Comment by jeffreykegler** [comment]
We could document that there is _always_ a lexer grammar associated with
any structural grammar.  If there is a `lexer` adverb, its value is the
name of the lexcial grammar.
Otherwise, if that grammar is `g1`, it is `l0`.  Otherwise, it is
`xyz_lex`, where `xyz` was the name of the original grammar.

I also wonder we should keep the "traditional" default names `g1` and
`l0`.  I didn't think they would be one of my naming successes, but they
instantly caught on and seemed to be quite intuitive.  So should we
continue with success, or have it run its course?

**Comment by rns** [comment]
We will see, I think. :) That grammar start rule thing seems to affect many aspects of the LUIF at once, so I'm still wrapping my head around that now (work is also creeping in) so I'll let it stay a bit.

**Comment by rns** [comment]
start rule: first cut 5596048

**Comment by rns** [comment]
re https://github.com/rns/kollos-luif-doc/issues/9#issuecomment-101743853 -- I can see the point, we can do, e.g.

```
:grammar:= start = <start symbol>, lexer = <lexical grammar name>
```

If not for the start symbol, start rule can be needed for `g1` to specify a lexical grammar other than `l0`?

**Comment by jeffreykegler** [comment]
Yes, it might make sense to restrict the `lexer` adverb to the start rule, so that it becomes the way of explicitly declaring a start rule, if desired.

**Comment by rns** [comment]
re `xyz_lex`, where `xyz` was the name of the original grammar -- bed07f0

re `g1` and `l0` -- can't really see anything against ‘continue with success’.

**Comment by jeffreykegler** [comment]
The latest version on the start rule language looks like it's headed in the
right direction.  I may have some nits after a careful reading, but the
general idea and most of the details are there.

**Comment by rns** [comment]
Great, so I'll label this one `pending review`.

**Comment by jeffreykegler** [comment]
Your choice.  You can also close it.  I expect that we'll be repeatedly
re-reading and revising the reference manual for some time, and it may be
best to have open Github issues only when there is some specific issue in
the way of progress.

Right now we are doing a kind of "inner circle" review, where the inner
circle is just you and I.  It'd be nice if it was a bit wider, but that's
OK.

Once we've done the "inner circle" review and have a draft, I want to put
it out to the Marpa community, where every issue can potentially be
re-opened and re-argued.

And, of course, once that is finished, we'll still be in alpha, so thing
can _still_ change.

**Comment by rns** [comment]
re keeping Github issues opened (and labelled appropriately) for subsequent broader review by the community -- sounds good. They'd provide context and entry points for future reviewers.
